nft/ethereum_payment_obligation.go

```diff
@@ -275,6 +416,23 @@ func getStoredNFT(nfts []*coredocumentpb.NFT, registry []byte) *coredocumentpb.N
 	return nil
 }
 
+// getRoleForAccount returns the matching role and index of the account
+func getRoleForAccount(cd *coredocumentpb.CoreDocument, roleName string, id identity.CentID) (idx int, role *coredocumentpb.Role) {
+	sha := sha256.New()
```

Comment:
We shouldn't default to sha256 here but require the client to specify the hash of the role, not the name. I don't think it should be a requirement that you use a specific hash function for roles.

Reply:
I have used sha256 because we haven't defined how the role label is filled out ATM. Would users will give 32 byte key for role? My understanding from the discussion with @pstehlik is that we might take a possible name like "supplier" as the role label. In that case, we would need to covert it to a 32 byte word

Reply:
The idea is that this is a convention we suggest people to follow but it isn't hard coded. We can't reverse this hash so when we would return the roles to the user we couldn't show the preimage of the hash which leads to odd behavior anyway. 

The idea is that in the future we can allow users to setup "labels" for these identifiers but this would work on a higher level in the node.